src/main/java/io/fabric8/maven/docker/util/Task.java

```diff
@@ -0,0 +1,15 @@
+package io.fabric8.maven.docker.util;
+
+import io.fabric8.maven.docker.access.DockerAccessException;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+
+/**
+ * Represents a generic task to be executed on a object.
+ */
+public interface Task<T> {
+
+    void execute(T object) throws DockerAccessException, MojoExecutionException, MojoFailureException;
```

Comment:
Would not throw Mojo exceptions here. Actually the more we can separate maven related and non-maven related code already, the easier will be the next step to separate those code (whether we will actually do it is another question ;-). But I don't think these exceptions are needed here (at least not in the single implementation in d-m-p. Maybe some f-m-p implementations ?)

The ideal goal would be to have all services independent of Maven and the parts which are needed from Maven then hidden behind interfaces (where we then implement those as adapter in the "maven" layer). But its more a long term goal (still not sure whether it's worth the effort)